PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp

```diff
@@ -1825,9 +1826,63 @@ void ScintillaEditView::restoreCurrentPos()
 		execute(SCI_SETXOFFSET, pos._xOffset);
 	}
 	execute(SCI_CHOOSECARETX); // choose current x position
-
 	int lineToShow = static_cast<int32_t>(execute(SCI_VISIBLEFROMDOCLINE, pos._firstVisibleLine));
-	scroll(0, lineToShow);
+	execute(SCI_SETFIRSTVISIBLELINE, lineToShow);
+	if (isWrap())
+	{
+		// Enable flag 'positionRestoreNeeded' so that function restoreCurrentPosAfterPainted get called
+		// once scintilla send SCN_PAITED notification
+		_positionRestoreNeeded = true;
+	}
+	_restorePositionRetryCount = 0;
+
+}
+
+void ScintillaEditView::restoreCurrentPosAfterPainted()
+{
+	static int32_t restoreDone = 0;
+	Buffer * buf = MainFileManager.getBufferByID(_currentBufferID);
+	Position & pos = buf->getPosition(this);
+
+	++_restorePositionRetryCount;
+
+	if (_restorePositionRetryCount > 8)

```

Comment:
Why do we need to retry 7 times?

Reply:
See my previous comment

Reply:
OK, I see. So please add comment on this magic number with your above explanation.